wallet/src/utils/cryptography/mnemonics.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import bip39 from 'bip39-light';
+import nacl from 'tweetnacl';
+
+import type { Ed25519KeypairData } from '@mysten/sui.js';
+
+/**
+ * Generate a 12-word random mnemonic and keypair using crypto.randomBytes
+ * under the hood, defaults to 128-bits of entropy.
+ * @returns a tuple of mnemonic and keypair. The mnemonics is a 12-word string
+ * split by spaces.
+ */
+export function generateMnemonicsAndKeypair(): [string, Ed25519KeypairData] {
+    const mnemonics = bip39.generateMnemonic();
+    return [mnemonics, getKeypairFromMnemonics(mnemonics)];
+}
+
+/**
+ * Derive public key and private key from the Mnemonics
+ * @param mnemonics a 12-word seed phrase
+ * @returns public key and private key
+ */
+export function getKeypairFromMnemonics(mnemonics: string): Ed25519KeypairData {
+    const seed = bip39.mnemonicToSeed(normalizeMnemonics(mnemonics));
+    return nacl.sign.keyPair.fromSeed(
+        // keyPair.fromSeed only takes a 32-byte array where `seed` is a 64-byte array
```

Comment:
Do we have any reference (from BIP39 RFC or a reference crypto key impl) that we typically use the first 32 bytes from the seed?

Reply:
> I feel that this functionality could go to the SDK, any reasons not putting it there?

@pchrysochoidis good question. I initially put this in the SDK but end up putting in the wallet for a few reasons:
1. This is only relevant to wallet applications, which is not the primary audience of the SDK(we will have 1000+ DApps but probably less than 10 JS based wallet
2. We are still iterating on this(i.e., implementing a V0 wallet) so don't want to mislead people who are using the SDK. We can consider moving this to the SDK once we feel very confident about our implementation

Reply:
I think it's fine to accept this method as it's already an output of PBKDF2 w SHA512. EdDSA does truncation already of sha512 outputs in its internal operations, and thus we're at least equally safe.